### ISSUE TITLE
open() Context Manager reports error

### ISSUE THREAD

**Issue by lovetox** [microsoft/pyright#4370]
**Describe the bug**
```python
from typing import Any

filename: Any = 'asd'

with open(filename, 'wb') as file:
    print(file)
```

the mode argument here is important, and also the type Any of the filename. For example a mode of `w` does not trigger the error

Errors:
```
   17:11  error   "path" is not defined ​Pyright:
   17:11  error   Argument type is unknown ​Pyright:
                    Argument corresponds to parameter "file" in function "open"
   17:26  error   Type of "file" is unknown ​Pyright:
   18:11  error   Argument type is unknown ​Pyright:
                    Argument corresponds to parameter "values" in function "print"
```

**Expected behavior**
Im not sure what is correct here, my gut feeling says a type Any should not raise a TypeError when passed to a method

**Additional context**
Im using pyright 1.1.285

The Problem does not occur with pyright 1.1.284

**Comment by erictraut** [comment]
This is intended behavior. You apparently have strict-mode type checking enabled because the errors you reported are part of the `reportUnknownVariableType` diagnostic check, which is disabled by default.

Pyright 1.1.285 changed the way it resolves overloads when the overload match is ambiguous due to an `Any` argument. In such a case, it evaluates the call as `Unknown` because there's no way for it to know which overload should be used.

If you don't want to see type errors related to `Unknown` variable usage, you can disable strict-mode type checking in your code.
